src/rgw/rgw_role.h

```diff
@@ -59,7 +59,6 @@ class RGWRole
     tenant(std::move(tenant)) {
     if (this->path.empty())
       this->path = "/";
-    extract_name_tenant(this->name);
```

Comment:
Please don't remove this parsing - this is used for parsing of role names passed in by radosgw-admin command, which allows to pass in role names as 'tenant$role'. Instead you can check if the tenant is populated then it should match with the extracted tenant name, else return an error.

Reply:
code has updated